packages/backend/src/open_payments/payment/incoming/service.test.ts

```diff
@@ -421,7 +500,8 @@ describe('Incoming Payment Service', (): void => {
       getPage: (pagination: Pagination) =>
         incomingPaymentService.getPaymentPointerPage(
           paymentPointerId,
-          pagination
+          pagination,
+          client ? grant.clientId : undefined
```

Comment:
In order to test that when `clientId` is specified, incoming payments with a different or no `clientId` is returned, you could create a few other incoming payments in `beforeEach`.
We probably should have already been doing that to test that incoming payments for different payment pointers are also excluded.
Maybe the extra incoming payments should include same payment pointer, different `clientId` and vice versa.

Reply:
Done for grants. Issue for payment pointers [here](https://github.com/interledger/rafiki/issues/602).